### DISCUSSION TITLE
All Inertia requests must receive a valid Inertia response, however a plain JSON response was received.

### DISCUSSION THREAD

**Discussion by cristian807** [inertiajs/inertia#1071] [Help]
Hello!
I have a problem with inertia when rendering the pages, it is showing me the data in the modal instead of rendering the page with the data in the table, taking into account that I uploaded the application to a shared server (cpanel) and I am using laravel 8.

**Comment by ajnsn** [comment]
Hey @cristian807 
What you describe means that the response isn't an Inertia response. Did you register Inertia's middleware as described [here](https://inertiajs.com/server-side-setup#middleware)? Is the server part different, means does it work locally?

**Reply by cristian807** [reply]
hello!
@ajnsn  Thank you for responding, the middleware is enabled, I have tried several methods that I have found on the internet and I have not been able to solve my problem.

**Reply by ajnsn** [reply]
Hey @cristian807 - ah good. Can you show how you are building the response, are you returning an Inertia response like [described here](https://inertiajs.com/responses)? Is the website publicly accessible somewhere?

**Reply by cristian807** [reply]
> hey @ajnsn, This is how I am returning the response. 
not accessible to the internet.

After that I have noticed that when I use the paginate() method it returns all the data in the modal instead of rendering the view and sending the data.

if I replace the paginate() method with get() it works up to a certain amount of records
`<?php

namespace App\Http\Controllers;

use Inertia\Inertia;
use App\Models\Grupo;
use App\Models\Estado;
use App\Models\Jornada;
use App\Models\Periodo;
use App\Models\Programa;
use App\Models\Estudiante;

class EstudianteController extends Controller
{
    public function index(Request $request)
    {
        Gate::authorize('tieneAcceso','estudiantes.index');

        return Inertia::render('Estudiantes/index',[
            'estudiantes'=>Estudiante::withTrashed()
                ->with('programa','estado')
                ->filters($request->only('buscar','programa','estado','periodo'))
                ->orderBy('id','DESC')
                ->get(),
            'programas'=>Programa::all(),
            'estados'=>Estado::all(),
            'grupos'=>Grupo::all(),
            'periodos'=>Periodo::all(),
            'jornadas'=>Jornada::all(),
            'can'=>[
                'create'=>auth()->user()->can('tieneAcceso','estudiantes.create'),
                'show'=>auth()->user()->can('tieneAcceso','estudiantes.show'),
                'edit'=>auth()->user()->can('tieneAcceso','estudiantes.edit'),
                'destroy'=>auth()->user()->can('tieneAcceso','estudiantes.destroy'),
                'restore'=>auth()->user()->can('tieneAcceso','estudiantes.restore'),
                'forcedelete'=>auth()->user()->can('tieneAcceso','estudiantes.force-delete'),
            ],
        ]);
    }
}`

**Reply by ajnsn** [reply]
@cristian807 So it's only happening with a single render/component and all other pages work? When you revisit (e.g. clicking a link in the navigation) the affected page, could you please inspect your network tab and look for Inerita's XHR request. What is returned there?

You could always take a look at PingCRM, e.g. the [OrganizationsController](https://github.com/inertiajs/pingcrm/blob/master/app/Http/Controllers/OrganizationsController.php) to see how Jonathan and Claudio implemented pagination.
